app/lib/dartdoc/dartdoc_runner.dart

```diff
@@ -19,7 +28,68 @@ class DartdocRunner implements TaskRunner {
 
   @override
   Future<bool> runTask(Task task) async {
-    // TODO: implement doc generation and upload
-    return false; // no race detected
+    final tempDir =
+        await Directory.systemTemp.createTemp('pub-dartlang-dartdoc');
+    final pubCacheDir = await tempDir.resolveSymbolicLinks();
+
+    final pubEnv = new PubEnvironment(pubCacheDir: pubCacheDir);
+
+    try {
+      final pkgLocation =
+          await pubEnv.getLocation(task.package, version: task.version);
+      final pkgPath = pkgLocation.location;
+      await _runDartdoc(task, pkgPath);
+      await _writeMetadata(task, pkgPath);
+      // TODO: upload doc/api to the appropriate bucket
+    } finally {
+      await tempDir.delete(recursive: true);
+    }
+    return false; // no race detection
```

Comment:
explain the comment?

Reply:
The method should return true if there were a race of processing detected (e.g. in `analyzer`), false otherwise. Not sure how we want to comment on it in such case.